test/dnsregistrar/TestOffchainDNSResolver.js

```diff
@@ -0,0 +1,206 @@
+    )).to.be.revertedWith('OffchainLookup('
+      + '"' + offchainResolver.address + '", '
+      + '["https://localhost:8000/query"], '
+      + '"' + DNSGatewayInterface.encodeFunctionData('resolve', [dnsName, 16]) + '", '
+      + '"' + ethers.utils.id('resolveCallback(bytes,bytes)').slice(0, 10) + '", '
+      + '"' + ethers.utils.defaultAbiCoder.encode(['bytes', 'bytes'], [dnsName, callData]) + '"'
+      + ')'
+    );
+  })
+
+  function doResolveCallback(name, texts, callData) {
+    const proof = [
+      hexEncodeSignedSet(rootKeys(expiration, inception)),
+      hexEncodeSignedSet(testRrset(name, texts)),
+    ];
+    const response = ethers.utils.defaultAbiCoder.encode(
+      ["tuple(bytes, bytes)[]"],
+      [proof]
+    );
+    const dnsName = utils.hexEncodeName(name);
+    const extraData = ethers.utils.defaultAbiCoder.encode(
+      ["bytes", "bytes"],
+      [dnsName, callData]
+    );
+    return offchainResolver.resolveCallback(response, extraData);
+  }
+
+  it('handles calls to resolveCallback() with valid DNS TXT records containing an address', async function() {
+    const name = "test.test";
+    const testAddress = '0xfefeFEFeFEFEFEFEFeFefefefefeFEfEfefefEfe';
+    await ownedResolver.setAddr(namehash.hash(name), testAddress);
+    const pr = await PublicResolver.at(offchainResolver.address);
+    const callData = pr.contract.methods['addr(bytes32)'](namehash.hash(name)).encodeABI();
+    const result = await doResolveCallback(name, [`ENS1 ${ownedResolver.address}`], callData);
+    expect(ethers.utils.defaultAbiCoder.decode(['address'], result)[0]).to.equal(testAddress);
+  })
+
+  it('handles calls to resolveCallback() with extra data and a legacy resolver', async function() {
+    const name = "test.test";
+    const testAddress = '0xfefeFEFeFEFEFEFEFeFefefefefeFEfEfefefEfe';
+    await ownedResolver.setAddr(namehash.hash(name), testAddress);
+    const pr = await PublicResolver.at(offchainResolver.address);
+    const callData = pr.contract.methods['addr(bytes32)'](namehash.hash(name)).encodeABI();
+    const result = await doResolveCallback(name, [`ENS1 ${ownedResolver.address} blah`], callData);
+    expect(ethers.utils.defaultAbiCoder.decode(['address'], result)[0]).to.equal(testAddress);
+  })
+
+  it('handles calls to resolveCallback() with valid DNS TXT records containing a name', async function() {
+    // Configure dnsresolver.eth to resolve to the ownedResolver so we can use it in the test
+    await root.setSubnodeOwner(ethers.utils.id('eth'), accounts[0]);
+    await ens.setSubnodeOwner(namehash.hash('eth'), ethers.utils.id('dnsresolver'), accounts[0]);
+    await ens.setResolver(namehash.hash('dnsresolver.eth'), ownedResolver.address);
+    await ownedResolver.setAddr(namehash.hash('dnsresolver.eth'), ownedResolver.address);
+
+    const name = "test.test";
+    const testAddress = '0xfefeFEFeFEFEFEFEFeFefefefefeFEfEfefefEfe';
+    await ownedResolver.setAddr(namehash.hash('test.test'), testAddress);
```

Comment:
can you just put `await ownedResolver.setAddr(namehash.hash(name), testAddress);` for the consistency unless this line is doing something different to other test?